### DISCUSSION TITLE
Annotations for editor

### DISCUSSION THREAD

**Discussion by msmcculley** [ppy/osu#19111] [Ideas]
Hello!

I was thinking about this for some time and I think it might be useful or good for mappers and modders alike! Sometimes when mapping ideas, patterns, and the like are fleeting and having to switch to a 3rd-party software to jot down the idea can be time-consuming and... quite honestly, annoying. 

The idea is that bookmarks can become bookmarks + annotations, what I mean by this is that when the player adds a bookmark in the editor a small little "notepad" type window would pop up to allow a brief jot down of an idea. For a lack of better words, it would allow for the mapper to leave an idea down and not have to worry about trying to remember what they wanted to do at this moment/section of the map. 

Now, you may be wondering about the mappers can place HUNDREDS of bookmarks down and at very close instances of time, I think a new tab (shown below) should open up a dedicated window/drop-down page that allows the mapper/user to manage the bookmark annotations

The single instance annotation would look something like: ![annotation-bookmark-single](https://user-images.githubusercontent.com/90729492/178792562-d7b6b822-e64a-4db3-b7d8-ae7819072218.png)

The tab to the dedicated window/annotation page would look something like:

![bookmark-dedicated-tab](https://user-images.githubusercontent.com/90729492/178791193-32f42d22-6efd-4714-9827-7e311f8f187b.jpg)

And the actual annotation page could look something like the `verify` page, just formatted for bookmarks: 

![dedicated-annotation-bookmark-page](https://user-images.githubusercontent.com/90729492/178792975-bf7c275b-c7f8-43ef-882b-12a0ed71378b.png)

**Comment by peppy** [comment]
Definitely something we'll add along with bookmarks. Although we still need a textbox UI control for these kinds of things.

**Comment by Soetch** [comment]
Maybe could it be a button with an on-click popup, attached to the tempo lines ? 
![image](https://user-images.githubusercontent.com/70801309/179240838-26fd6adf-94dd-47e0-8df4-27fdf421b2ca.png)

![image](https://user-images.githubusercontent.com/70801309/179241348-9a8f15de-4ec3-45c5-8d96-168bb5cf1027.png)

**Reply by msmcculley** [reply]
Hmm, I think this may depend on what devs may think would be "too much" for any given annotation. In my mind, the pop-up text would continue to grow in size as the user typed: as any fledged-out idea could be paragraphs long in certain languages and fonts. I do think that attaching it to the timeline where the bookmark is placed is better for user readability. Along with this, a little "+" would show up on the timeline to add an annotation or ">" "v" to review an already placed annotation.

**Reply by Soetch** [reply]
### Just saying this because the zone you showed was kinda too much, to my eyes !
![image](https://user-images.githubusercontent.com/70801309/179292393-46533783-d1d4-45d5-b104-d9b6131a1609.png)

### I think it has to be in the timeline too, but maybe it can be a thing like this : 
![image](https://user-images.githubusercontent.com/70801309/179293859-7bac2ca2-abae-4d18-83f5-baedb3951a94.png)


(ps : sorry, I love making drawings😭)

**Reply by msmcculley** [reply]
The only issue I could see with this is that if they have multiple annotations within those time periods (and just because people can be shitheads and like to limit test the program, which is why I suggested a ">" and "v" drop-down type of button on the bookmark/timeline where the annotation is. 

I don't really see how or why markdown would be needed? The entire purpose of this is supposed to be a note or a quick comment. I suggested an an-ever-growing textbox because languages like Russian, Greek, and other non-latinized languages can sometimes have words that are super long. Modding and mapping doesn't really have a use for markdown. There should be no reason you need `### 00:05:00`
